### PR TITLE
Generate tests in header-parser with Vanilla

### PR DESCRIPTION
**Pull request by JanKrivanek**

### CODE REVIEW

File: packages/header-parser/test/index.test.ts [hunk 1]
```diff
@@ -174,3 +185,741 @@ Expected one of: ["MIT","Apache-2.0"]}`,
...
+    const result = validatePackageJson("mylib", makePkg({ nonNpm: true }), []);
+    expect(result).toEqual(
+      expect.arrayContaining([
+        expect.stringContaining('has missing "nonNpmDescription"'),
+      ]),
+    );
+  });
+
+  it("rejects nonNpm: true with non-string nonNpmDescription", () => {
+    const result = validatePackageJson("mylib", makePkg({ nonNpm: true, nonNpmDescription: 123 }), []);
```
Comment (sandersn) [inline packages/header-parser/test/index.test.ts:247]: good coverage of edge cases

File: packages/header-parser/test/index.test.ts [hunk 2]
```diff
@@ -174,3 +185,741 @@ Expected one of: ["MIT","Apache-2.0"]}`,
...
+      name: "@types/mylib",
+      version: "1.0.9999",
+      devDependencies: { "@types/mylib": "workspace:." },
+      owners: [{ name: "Test User", githubUsername: "testuser" }],
+      projects: ["https://example.com"],
+      ...overrides,
+    };
+  }
+
+  it("rejects missing projects", () => {
```
Comment (sandersn) [inline packages/header-parser/test/index.test.ts:324]: this is one that's basic enough that I had trouble remembering what it was testing. So it's a good check-your-assumptions case.

File: packages/header-parser/test/index.test.ts [hunk 3]
```diff
@@ -1,4 +1,15 @@
-import { validatePackageJson, makeTypesVersionsForPackageJson, License, getLicenseFromPackageJson } from "../src";
+import {
```
Comment (sandersn) [inline packages/header-parser/test/index.test.ts:1]: don't reformat these; this repo is formatted by prettier (or dprint? can't remember)

**Review by sandersn — COMMENTED** [review]
From memory, this looks like complete coverage of the properties we neglected to test. Those are low-value but it's nice to have  better coverage.

File: packages/header-parser/test/index.test.ts (hunk 3 above)
Comment (sandersn) [reply packages/header-parser/test/index.test.ts:1]: oh that's because there are more things imported now
